packages/miniflare/test/index.spec.ts

```diff
@@ -1713,3 +1740,199 @@ test("Miniflare: prohibits invalid wrapped bindings", async (t) => {
+								} catch (e) {
+									return new Response(String(e), { status: 500 });
+								}
+							}
+						}
+						`,
+					},
+				],
+			},
+		],
+	});
+	t.teardown(() => mf.dispose());
+
+	let res = await mf.dispatchFetch("http://localhost/a");
+	t.is(await res.text(), "acd");
+
+	// Check fallback service ignored if not explicitly enabled
+	res = await mf.dispatchFetch("http://localhost/b");
+	t.is(res.status, 500);
+	t.is(await res.text(), 'Error: No such module "virtual/a.mjs".');
+});
+
+test.serial(
+	"Miniflare: respects rootPath for path-valued options",
+	async (t) => {
+		const tmp = await useTmp(t);
+		const aPath = path.join(tmp, "a");
+		const bPath = path.join(tmp, "b");
+		await fs.mkdir(aPath);
+		await fs.mkdir(bPath);
+		await fs.writeFile(path.join(aPath, "1.txt"), "one text");
+		await fs.writeFile(path.join(aPath, "1.bin"), "one data");
+		await fs.writeFile(path.join(aPath, "add.wasm"), ADD_WASM_MODULE);
+		await fs.writeFile(path.join(bPath, "2.txt"), "two text");
+		await fs.writeFile(path.join(tmp, "3.txt"), "three text");
+		const mf = new Miniflare({
+			rootPath: tmp,
+			kvPersist: "kv",
+			workers: [
+				{
+					name: "a",
+					rootPath: "a",
+					routes: ["*/a"],
+					textBlobBindings: { TEXT: "1.txt" },
+					dataBlobBindings: { DATA: "1.bin" },
+					wasmBindings: { ADD: "add.wasm" },
+					// WASM bindings aren't supported by modules workers
+					script: `addEventListener("fetch", (event) => {
+						event.respondWith(Response.json({
+							text: TEXT,
+							data: new TextDecoder().decode(DATA),
+							result: new WebAssembly.Instance(ADD).exports.add(1, 2)
+						}));
+					});`,
+				},
+				{
+					name: "b",
+					rootPath: "b",
+					routes: ["*/b"],
+					textBlobBindings: { TEXT: "2.txt" },
+					sitePath: ".",
+					script: `addEventListener("fetch", (event) => {
+						event.respondWith(Response.json({
+							text: TEXT,
+							manifest: Object.keys(__STATIC_CONTENT_MANIFEST)
+						}));
+					});`,
+				},
+				{
+					name: "c",
+					routes: ["*/c"],
+					textBlobBindings: { TEXT: "3.txt" },
+					kvNamespaces: { NAMESPACE: "namespace" },
+					modules: true,
+					script: `export default {
+						async fetch(request, env, ctx) {
+						 	await env.NAMESPACE.put("key", "value");
+							return Response.json({ text: env.TEXT });
+						}
+					}`,
+				},
+			],
+		});
+		t.teardown(() => mf.dispose());
+
+		let res = await mf.dispatchFetch("http://localhost/a");
+		t.deepEqual(await res.json(), {
+			text: "one text",
+			data: "one data",
+			result: 3,
+		});
+		res = await mf.dispatchFetch("http://localhost/b");
+		t.deepEqual(await res.json(), {
+			text: "two text",
+			manifest: ["2.txt"],
+		});
+		res = await mf.dispatchFetch("http://localhost/c");
+		t.deepEqual(await res.json(), {
+			text: "three text",
+		});
+		t.true(existsSync(path.join(tmp, "kv", "namespace")));
+
+		// Check persistence URLs not resolved relative to root path
+		await mf.setOptions({
```

Comment:
How do these calls to `setOptions()` "check" things?
Are you saying that they will throw if things are not alright?